File: 73-set-matrix-zeroes/set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """

        arr : List[Tuple[int,int]] = []

        for i in range (0,len(matrix)):
            for j in range (0, len(matrix[0])):
                if(matrix[i][j] == 0):
                    arr.append((i,j))
        

        for pos in arr:
            ipos , jpos = pos

            for i in range (0,len(matrix)):
                matrix[i][jpos] = 0

            for j in range (0, len(matrix[0])):
                matrix[ipos][j] = 0
        
    
        

File: 73-set-matrix-zeroes/set-matrix-zeroes.ts
/**
 Do not return anything, modify matrix in-place instead.
 */
function setZeroes(matrix: number[][]): void {

    let arr :[number,number][] = [];

    for(let i:number = 0;i<matrix.length;i++)
    {
        for(let j:number = 0;j<matrix[0].length;j++)
        {
            if(matrix[i][j] === 0)
            {
                arr.push([i,j]);
            }
        }
    }

    for(let pos of arr)
    {
        let [ipos , jpos] = pos;

        for(let i:number = 0;i<matrix.length;i++)
        {
            matrix[i][jpos] = 0;
        }

        for(let j:number = 0;j<matrix[0].length;j++)
        {
            matrix[ipos][j] = 0;
        }
    }
};